src/wallet/transaction.h

```diff
@@ -232,8 +211,10 @@ class CWalletTx
             setConfirmed();
         }
 
-        ReadOrderPos(nOrderPos, mapValue);
-        nTimeSmart = mapValue.count("timesmart") ? (unsigned int)atoi64(mapValue["timesmart"]) : 0;
+        const auto it_op = mapValue.find("n");
+        nOrderPos = (it_op != mapValue.end()) ? atoi64(it_op->second) : -1;
+        const auto it_ts = mapValue.find("timesmart");
+        nTimeSmart = (it_ts != mapValue.end()) ? (unsigned int)atoi64(it_ts->second) : 0;
```

Comment:
Maybe use a named cast while touching this line instead of a C-style cast.

Reply:
Thanks, that seems reasonable, done.